### DISCUSSION TITLE
Multiple routes with the same component

### DISCUSSION THREAD

**Discussion by gondc** [posva/unplugin-vue-router#129] [Questions and Help] [answered]
Is there a way to have multiple routes using the same component file? without having to copy the file to each page folder

Example: I have `[userId].vue` and I want to have two routes that render that file `/users/:userId` and `/clients/:userId`

And I don't want to have to copy the `[userId].vue` to every page folder, is a really bad solution

Bad solution:
```
src/pages/
├── users/
│   └── [userId].vue
└── clients/
    └── [userId].vue (copy of /pages/usrs/[userId].vue)
```

**Comment by posva** [comment, marked as answer]
Usually you would define an alias. If two routes are required, one could add an extra route with extendRoutes

**Comment by zhiyuanzmj** [comment]
You can try `definePage` macro:
``` vue
// clients/[userId].vue
<script setup lang="ts">
definePage({
   component: ()=>import('../users/[userId].vue')
})
</script>
```
